### ISSUE TITLE
Atom feed titles need to be more expressive

### ISSUE THREAD

**Issue by jbond23** [tootsuite/mastodon#1438]
A recent update (https://github.com/tootsuite/mastodon/pull/1343) changed Atom feeds so that all item titles read "New status by userid". This may work for federation but it's user unfriendly if you're reading the Atom in an Atom reader. I'd prefer to see something like the first 128 chars of an html_stripped version of the description. userid is already in the channel.title so that's redundant in the item. And "new entry" is obvious.

item.title is generally optional but lots of readers fail when it's missing. It's not really clear how title, summary and description should be used for source data that only has one field such as Mastodon and Twitter. So having a short version of the html stripped full text seems to be an acceptable compromise.

**Comment by sycom** [comment]
Absolutely agree, atom is about wide openness. Title could be
* all chars before a comma `,`, or any separator
* if none first 128 chars

**Comment by husimo** [comment]
Maybe it should be interesting to have the choice. Display the 128 first chars or "New status by userid". It's not user friendly at all currently.

**Comment by thiagomgd** [comment]
This should be an option, because nowadays many people are using RSS to integrate Mastodon with other apps by using IFTTT or similar services, and the fixed title can easily be used as a filter. But yes, I agree that it would be a nice option to have.

On another note, I would like replies titles to be `New reply by user` instead of `New status by user`. That way it would be easier to filter my toots, and my replies. :)

**Comment by MilkManzJourDaddy** [comment]
I see a use-case where, unfortunately due to this issue, a 3rd Party RSS Feed from Twitter™ is being used instead of an Atom Feed from Mastodon.  And, the client MastodonToTwitter is being used to post, so Mastodon **is** in the loop.  Example: This [Matrix, Mastodon Toot](https://mastodon.matrix.org/@matrix/228068) is the same content as this [Matrix, Twitter™ Tweet](https://twitter.com/matrixdotorg/status/945127067577978881).  In the [Matrix room for Mastodon](https://matrix.to/#/#Mastodon:disroot.org), [#Mastodon:disroot.org](https://riot.im/app/#/room/#Mastodon:disroot.org), (read-only via: https://view.matrix.org/alias/%23Mastodon:disroot.org ), we have a feed from [@Mastodon@mastodon.social](https://mastodon.social/@Mastodon)`.atom`, so we see this issue.  We all have limited time and we would rather have an idea of the content of the dozens of links thrown at us before we click, otherwise they are ignored.  One possible solution mentioned was https://gitlab.com/ilpianista/FeedToot •but it would be better to have a wholistic approach within Mastodon itself.

**Comment by mike-burns** [comment]
@husimo @thiagomgd why would it be an option? What's the use case for toggling the useful title off?

**Comment by albjeremias** [comment]
Please don't break RSS... its such a great invention... how can this be fixed? I could try to make a pull request...

**Comment by mike-burns** [comment]
> how can this be fixed? I could try to make a pull request...

That is exactly how. Start in `app/models/status.rb` and `spec/fixtures/requests/activitypub-feed.txt`. I haven't looked into this more but that's where I'd start digging.

No one is taking RSS away. We'd like the title to be better.

**Comment by Gargron** [comment]
We don't have and never had RSS :thinking:

**Comment by albjeremias** [comment]
@Gargron you are right.. i meant atom.. where you can use the link: https://yourinstance/@user.atom
as for atom, mastodon is breaking the federation between websites, which was what atom/rss/xml feeds were already doing for a long time... so would be really awesome to have this working in a correct/standard way.

**Comment by mike-burns** [comment]
@albjeremias sorry, I don't understand. This ticket is for changing the title used in the atom feeds -- a cosmetic change. Are you reporting a different bug? Is something broken?

**Comment by albjeremias** [comment]
@mike-burns the title and the content of the RSS. which means it breaks the essence of RSS/Atom, and make mastodon useless for many rss clients/bots/networks, because you have to click to see the content.

**"RSS removes the need for the user to manually check the website for new content. Instead, their browser constantly monitors the site and informs the user of any updates."** https://en.wikipedia.org/wiki/RSS

**Comment by albjeremias** [comment]
@mike-burns im sorry, you are correct! that is exactly the problem the title of the mastodon .. the other which is also related is that the link when its a reply gives access to a json blob on the browser.!

I can't find an issue about this second issue.. should i create it?
I get this link:
https://mastodon.social/users/FuckOffGoogle/statuses/99891165779679992/activity

and expected this:
https://mastodon.social/users/FuckOffGoogle/statuses/99891165779679992 

without /activity

**Comment by Gargron** [comment]
This is what I mean when I say the Atom feeds were never meant for end-users, they're for OStatus federation. I no longer feel like this issue is warranted. There's probably an open issue about adding RSS instead

**Comment by jbond23** [comment]
How does an expressive title make the feed less suitable for OStatus federation? Is item.title actually used in OStatus Federation?

I have to say that I'm disappointed with the reason given for closing the issue. I have a use case for reading the atom feed and then displaying it in a sidebar of a blog. I'm having to recreate an abstract of the toot to display where it would be simpler and easier to just display an expressive title. 

If there's a use case for RSS as a user friendly syndication format, then there's the same use case for an Atom representation. And without getting into the religious atom vs rss wars, Atom is a superset of RSS.

**Comment by albjeremias** [comment]
@jbond23 it seems so... not even the links make sense, they point to a json blob, if the atom is changed the OStatus probably would break.

I guess there is already a PR... checkout it out in here:
 https://github.com/tootsuite/mastodon/issues/673

**Comment by nightpool** [comment]
that's not a pull request that's just an open issue

**Comment by albjeremias** [comment]
There were 2 pull requests but both did not solve this issue.. the titles of the rss feeds are still "New status by ajeremias" and "ajeremias shared a status by bernini@social.coop" instead of being the content inside the status on the title... but I guess this is more problematic to change... 

https://github.com/tootsuite/mastodon/pull/7259 & https://github.com/tootsuite/mastodon/pull/7436/files

**Comment by MilkManzJourDaddy** [comment]
I too am "disappointed with the reason given for closing the issue."  

I thought I presented a use-case above.  I could see some arbitrary work-flow thing where this issue was closed in favour of another **specificly listed issue**.  But probably does not cover that, and the tone is dismissive of user concerns.  I almost feel as if it's a canned response from Twitter™ ala "Thank you for your input.  We value your feedback. […but really don't give a crap!]".  

It's a sad day when we need to Reverse Engineer and find workarounds for F(L)OSS projects that are ***supposed*** to be for users.  But FWIW: https://github.com/RSS-Bridge/rss-bridge •
